torch/_refs/nn/functional/__init__.py

```diff
@@ -468,8 +469,7 @@ def softshrink(a: TensorLikeType, lambd: float = 0.5):
         lambd >= 0,
         lambda: f"lambda must be greater or equal to 0, but found to be {lambd}",
     )
-    ret = torch.where(a > lambd, a - lambd, 0)
-    return torch.where(a < -lambd, a + lambd, ret)
+    return torch.where(torch.abs(a) > lambd, a - torch.sign(a) * lambd, 0)
```

Comment:
I think it was more readable before tbh

Reply:
Agree. This rewriting allows to make it "composite implicit" via the Autograd key. 

The issues is that the derivative of `torch.where(c, a, b)` wrt a is given by `torch.where(c, a, 0)`, and wrt b is given by `torch.where(~c, 0, b)`. Now, if you have a function where both `a`, `b` are functions of a variable `x` (like here before), you end up generating a derivative as:

```python
tmp1 = tl.where(c, a, 0)
tmp2 = tl.where(~c, 0, b)
return tmp1 + tmp2
```
while you could simply write
```python
return tl.where(a, b)
```
This rewrite puts all the `a`'s on one side, so that we generate just one where in the backward rather than two. As in says in the OP, the efficiency of the forward/backward should be comparable or better than the one before I reckon.

Reply:
It'd be cool to be able to do that rewrite in inductor, but we don't have the code to do it yet.

Reply:
Fair enough, maybe worth a comment though?